spec/javascripts/stimulus/vertical_ellipsis_menu_controller_test.js

```diff
@@ -0,0 +1,55 @@
+/**
+ * @jest-environment jsdom
+ */
+
+import { Application } from "stimulus";
+import vertical_ellipsis_menu_controller from "../../../app/webpacker/controllers/vertical_ellipsis_menu_controller";
+
+describe("VerticalEllipsisMenuController test", () => {
+  beforeAll(() => {
+    const application = Application.start();
+    application.register("vertical-ellipsis-menu", vertical_ellipsis_menu_controller);
+  });
+
+  beforeEach(() => {
+    document.body.innerHTML = `
+      <div data-controller="vertical-ellipsis-menu" id="root">
+        <div data-action="click->vertical-ellipsis-menu#toggle" id="button">...</div>
+        <div data-vertical-ellipsis-menu-target="content" id="content">
+          
+        </div>
+      </div>
+    `;
+  });
+
+  it("add show class to content when toggle is called", () => {
+    const button = document.getElementById("button");
+    const content = document.getElementById("content");
+
+    expect(content.classList.contains("show")).toBe(false);
+    button.click();
+    expect(content.classList.contains("show")).toBe(true);
+  });
+
+
+  it("remove show class from content when clicking button", () => {
+    const button = document.getElementById("button");
+    const content = document.getElementById("content");
```

Comment:
Gaetan taught me that these variables can be assigned in beforeEach to remove duplication (similar to rspec's `let!`).
https://github.com/openfoodfoundation/openfoodnetwork/pull/11509#discussion_r1328441837

In fact, I found I didn't even need the initial `let` that he mentioned. Just moving it to beforeEach worked.
https://github.com/openfoodfoundation/openfoodnetwork/pull/11509/files#diff-e2b6d56ae1511dc914a8a816c75e5a707679378ecd28340d001adf25fde5a468R48-R73

Reply:
Oh yeah, much more easier. Super strange about this `const` assigned in another block but still available, but anyway...

Reply:
Yeah I don't understand, but am glad I don't need to 🙈